src/matrix/matrixfdist.jl

```diff
@@ -0,0 +1,143 @@
+"""
+```julia
+MatrixFDist(n1, n2, B)
+
+n1::Real  degrees of freedom (greater than p - 1)
+n2::Real  degrees of freedom (greater than p - 1)
+B::PDMat  p x p scale
+```
+The [matrix *F*-Distribution](https://projecteuclid.org/euclid.ba/1515747744)
+(sometimes called the matrix beta type II distribution) generalizes the
+*F*-Distribution to ``p\\times p`` real, positive definite matrices ``\\boldsymbol{\\Sigma}``.
+If ``\\boldsymbol{\\Sigma}\\sim MF_{p}(n_1/2,n_2/2,\\mathbf{B})``, then its probability density function is
+
+```math
+f(\\boldsymbol{\\Sigma} ; n_1,n_2,\\mathbf{B}) =
+\\frac{\\Gamma_p(\\frac{n_1+n_2}{2})}{\\Gamma_p(\\frac{n_1}{2})\\Gamma_p(\\frac{n_2}{2})}
+|\\mathbf{B}|^{n_2/2}|\\boldsymbol{\\Sigma}|^{(n_1-p-1)/2}|\\mathbf{B}+\\boldsymbol{\\Sigma}|^{-(n_1+n_2)/2}.
+```
+
+If the joint distribution ``p(\\boldsymbol{\\Psi},\\boldsymbol{\\Sigma})=p(\\boldsymbol{\\Psi})p(\\boldsymbol{\\Sigma}|\\boldsymbol{\\Psi})``
+is given by
+
+```math
+\\begin{align*}
+\\boldsymbol{\\Psi}&\\sim W_p(n_1, \\mathbf{B})\\\\
+\\boldsymbol{\\Sigma}|\\boldsymbol{\\Psi}&\\sim IW_p(n_2, \\boldsymbol{\\Psi}),
+\\end{align*}
+```
+
+then the marginal distribution of ``\\boldsymbol{\\Sigma}`` is
+``MF_{p}(n_1/2,n_2/2,\\mathbf{B})``.
+"""
+struct MatrixFDist{T <: Real, TW <: Wishart} <: ContinuousMatrixDistribution
+    W::TW
+    n2::T
+    logc0::T
+end
+
+#  -----------------------------------------------------------------------------
+#  Constructors
+#  -----------------------------------------------------------------------------
+
+function MatrixFDist(n1::Real, n2::Real, B::AbstractPDMat)
+
+    p = dim(B)
+
+    n1 > p - 1 || throw(ArgumentError("first degrees of freedom must be larger than $(p - 1)"))
+    n2 > p - 1 || throw(ArgumentError("second degrees of freedom must be larger than $(p - 1)"))
+
+    logc0 = matrixfdist_logc0(n1, n2, B)
+    T = Base.promote_eltype(n1, n2, logc0, B)
+    prom_B = convert(AbstractArray{T}, B)
+    W = Wishart(T(n1), prom_B)
+
+    MatrixFDist{T, typeof(W)}(W, T(n2), T(logc0))
+
+end
+
+MatrixFDist(n1::Real, n2::Real, B::Union{AbstractMatrix, LinearAlgebra.Cholesky}) = MatrixFDist(n1, n2, PDMat(B))
+
+#  -----------------------------------------------------------------------------
+#  REPL display
+#  -----------------------------------------------------------------------------
+
+ show(io::IO, d::MatrixFDist) = show_multline(io, d, [(:n1, d.W.df), (:n2, d.n2), (:B, d.W.S.mat)])
+
+#  -----------------------------------------------------------------------------
+#  Conversion
+#  -----------------------------------------------------------------------------
+
+function convert(::Type{MatrixFDist{T}}, d::MatrixFDist) where T <: Real
+    W = convert(Wishart{T}, d.W)
+    MatrixFDist{T, typeof(W)}(W, T(d.n2), T(d.logc0))
+end
+
+function convert(::Type{MatrixFDist{T}}, W::Wishart, n2, logc0) where T <: Real
```

Comment:
Shouldn't it be a subtype?:
```
function convert(::Type{<:MatrixFDist{T}}, W::Wishart, n2, logc0) where T <: Real
```

Reply:
Maybe? Frankly, for all of these I've just been naively copying how `convert` is written for `Wishart` and `InverseWishart`. 

https://github.com/JuliaStats/Distributions.jl/blob/dc2e589402fc8f814f29df40df933368badf849f/src/matrix/wishart.jl#L57-L64